src/monitor/node_metadata.c

```diff
@@ -438,15 +438,20 @@ GetPrimaryOrDemotedNodeInGroup(char *formationId, int32 groupId)
 		return primaryNode;
 	}
 
-	/* maybe we have a primary that is draining or has been demoted? */
+	/*
+	 * Maybe we have a primary that is draining or has been demoted?
+	 * In case there are more than one of those, choose the one that is
+	 * currently being demoted.
+	 */
 	foreach(nodeCell, groupNodeList)
 	{
 		AutoFailoverNode *currentNode = (AutoFailoverNode *) lfirst(nodeCell);
 
-		if (StateBelongsToPrimary(currentNode->reportedState))
+		if (StateBelongsToPrimary(currentNode->reportedState) &&
+			(IsBeingDemotedPrimary(primaryNode) == false ||
+			IsDemotedPrimary(currentNode) == false))
```

Comment:
I think we would generally use a slightly different style:

``` c
		if (StateBelongsToPrimary(currentNode->reportedState) &&
  			(!IsBeingDemotedPrimary(primaryNode) ||
  			!IsDemotedPrimary(currentNode)))
```

Reply:
Heh, I was debating as to which was preferable :) Will fix.